### ISSUE TITLE
Getting caret position within the HTML tags

### ISSUE THREAD

**Issue by Galiza** [KillerCodeMonkey/ngx-quill#216]
Here's the thing, I'm trying something which isn't working and apparently the quill does not support.

I want to get the caret position inside the text editor, though I've tried a couple of things that partially worked, meaning that, it worked IF I wasn't considering the HTML tags, for example:

12345 in plain text, and I want the caret after 1, it'll say that the position of the caret is in position 1, which is correct, although what I'm trying to achieve is to get the position of the caret, when there's HTML tags. For example:

<p><b>12</b>345</p> (with bold tag inbetween 12 and with p tag at the beggining and end of sentence), it will be broken because it doesn't count the HTML tags.

And I need also to insert an HTML template with tags in midst of it all, does anyone knows the solution or have the solution to this?

What I implemented so far was the above, which is partially working:

`if (window.getSelection && window.getSelection().getRangeAt) {
        const selectedObj = window.getSelection();
        let rangeCount = 0;
        const childNodes = selectedObj.anchorNode.parentNode.childNodes;
        for (let i = 0; i < childNodes.length; i++) {
          if (childNodes[i] === selectedObj.anchorNode) {
            break;
          }
          if (childNodes[i].outerHTML) {
            rangeCount += childNodes[i].outerHTML.length;
          } else if (childNodes[i].nodeType === 3) {
            rangeCount += childNodes[i].textContent.length;
          }
        }
        this.caretPosition = caretPosition.range.index + rangeCount;
      }`

Then I get the substring from the beggining to the caret position, insert the template, then get the rest of the substring using concat on it all, but it works fine using a plain text without the HTML tags, and I need the one with the HTML tags.

**Comment by Galiza** [comment]
The setSelection does not work in the context I am asking about, with the HTML tags. But thanks anyway.

**Comment by KillerCodeMonkey** [comment]
maybe you have to make sure the window get and setSelection is executed in an angular zone.

The last thing i would try.

**Comment by KillerCodeMonkey** [comment]
any updates?

**Comment by Galiza** [comment]
So, I've actually used a different approach, because in fact the quill doesn't allow you to find the caret in the right position with the HTML tags.

We used the solution from this [Stackoverflow]( https://stackoverflow.com/questions/44768980/insert-raw-html-code-to-quill) question. Meaning that, we find the position of the index, we then embed the HTML into the body of the text, so we use .replace() to remove all the iframe tags that this solution creates.
